source/Nevermore/UniqueParameterNameGenerator.cs

```diff
@@ -12,31 +12,33 @@ public interface IUniqueParameterNameGenerator
     internal class UniqueParameterNameGenerator : IUniqueParameterNameGenerator
     {
         readonly HashSet<string> assigned = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
-        
+
         public string GenerateUniqueParameterName(string parameterName)
         {
-            var original = Parameter.Normalize(parameterName);
-            var candidate = original;
-            var counter = 0;
-            
-            while (true)
+            lock (assigned)
```

Comment:
Critical section here because we want consuming code to be able to construct multiple queries concurrently.